plugins/module_utils/foreman_helper.py

```diff
@@ -195,7 +210,7 @@ def update_resource(self, resource, old_entity, entity_dict, check_missing, chec
                     volatile_entity[key] = entity_dict[key]
                     fields.append(key)
         if len(fields) > 0:
-            new_data = {'id': entity_id}
+            new_data = entity
```

Comment:
with that change, we can drop the use of `new_data` entirely and just use `entity` everywhere